packages/edit-post/src/components/header/fullscreen-mode-close/test/index.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import FullscreenModeClose from '../';
+
+jest.mock( '@wordpress/data/src/components/use-select', () => {
+	// This allows us to tweak the returned value on each test
+	const mock = jest.fn();
+	return mock;
+} );
+
+jest.mock( '@wordpress/core-data' );
+
+describe( 'FullscreenModeClose', () => {
+	describe( 'when in full screen mode', () => {
+		it( 'should display a user uploaded site icon if it exists', () => {
+			useSelect.mockImplementation( ( cb ) => {
+				return cb( () => ( {
+					isResolving: () => false,
+					isFeatureActive: () => true,
+					getCurrentPostType: () => {},
+					getPostType: () => true,
+					getEntityRecord: () => ( {
+						site_icon_url: 'https://fakeUrl.com',
+					} ),
+				} ) );
+			} );
+
+			const { getByAltText } = render( <FullscreenModeClose /> );
+			const siteIcon = getByAltText( 'site-icon' );
```

Comment:
Maybe a different kind of selector (e.g. class name?) could be more stable? ([See](https://github.com/WordPress/gutenberg/pull/22952/files#r448381310))

Reply:
This makes sense since alt text may be dynamic now. 👍